### PR TITLE
Create QUALITY_GUIDELINES.md

### PR DESCRIPTION
**Pull request by LorenzoBettini**
We refer to this file in the README, but it got lost during the transition to monorepo.

### CODE REVIEW

File: QUALITY_GUIDELINES.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+ 6. Changes must have unit tests.
+ 7. Good changes are the ones that do not cause or even minimize future effort, e.g., bug fixes and follow-up feature requests.
+ 8. (Xtext Committers Only) PRs with peer reviews are required for changes to existing code, Xtext core, and new features, except for trivial changes (typo, docs).
+
+## Source Code
+ 1. Readable and understandable source code is good source code.
+ 2. Keep the source code well-formatted, preferably with the Eclipse formatter.
+ 3. Put high-level comments in the code to explain your intention, solved use cases, design decisions, rejected solutions, etc. Do *not* duplicate what is obvious from the code itself. People will read code + documentation; therefore, documentation does not need to be self-contained.
+
+## API
```
Comment (cdietrich) [inline QUALITY_GUIDELINES.md:18]: should we add `@since`

Comment (LorenzoBettini) [reply QUALITY_GUIDELINES.md:18]: Right! I've updated the document accordingly.

File: QUALITY_GUIDELINES.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
+This page lists criteria that code, commits, and pull requests should fulfill.
```
Comment (szarnekow) [inline QUALITY_GUIDELINES.md:1]: Not sure I adhere to the standard below. But I like them.

File: QUALITY_GUIDELINES.md [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+## Commits and Pull Requests
+ 1. Commit messages should have a short message and a description. The description should explain: (a) What's changed. (b) What problem is fixed, or what feature does work now (c) What alternative solutions were discarded and why.
+ 2. Commit messages should refer to the Github Issue; if there is one, you can do so by prefixing your commit title with `[#issueTicketNr]`, e.g., `[#123]`. You can refer to a GitHub issue in another GitHub repository by prefixing the title of your commit with `[githubUser/githubRepo#issueTicketNr]`, e.g., `[eclipse/xtext#123]`.
```
Comment (szarnekow) [inline QUALITY_GUIDELINES.md:5]: Can this be enforced?

Comment (cdietrich) [reply QUALITY_GUIDELINES.md:5]: am not sure if this was a burden

File: QUALITY_GUIDELINES.md (hunk 2 above)
Comment (LorenzoBettini) [reply QUALITY_GUIDELINES.md:1]: We do the best we can ;)

File: QUALITY_GUIDELINES.md (hunk 3 above)
Comment (LorenzoBettini) [reply QUALITY_GUIDELINES.md:5]: I don't know if it can.

Comment (szarnekow) [reply QUALITY_GUIDELINES.md:5]: > am not sure if this was a burden

true.

**Review by cdietrich — APPROVED** [review]

**Pull request merged**